### PR TITLE
currency: Add Pair method IsAssociated

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

* Derives pair association by having a similar currency code between pairs: 
    * currency.NewPair(BTC, USDT).IsAssociated(currency.NewPair(BTC, USD)) returns true (BTC association)
    * currency.NewPair(XRP, USDT).IsAssociated(currency.NewPair(BTC, USD)) returns false (no association)
* Exports error 
* Add field pair in OrderParameters.

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: currency/pair_test.go [hunk 1]
```diff
@@ -1033,6 +1033,33 @@ func TestGetOrderParameters(t *testing.T) {
...
+		{Pair: NewPair(BTC, USDT), associate: NewPair(BTC, USDT), expectedResult: true},
+		{Pair: NewPair(USDT, BTC), associate: NewPair(BTC, USDT), expectedResult: true},
+		{Pair: NewPair(BTC, USDT), associate: NewPair(USDT, BTC), expectedResult: true},
+		{Pair: NewPair(BTC, USDT), associate: NewPair(XRP, USDT), expectedResult: true},
+		{Pair: NewPair(BTC, LTC), associate: NewPair(XRP, USDT), expectedResult: false},
+		{Pair: NewPair(MA, LTC), associate: NewPair(LTC, USDT), expectedResult: true},
+	}
+
+	for x := range testCases {
+		if testCases[x].Pair.IsAssociated(testCases[x].associate) != testCases[x].expectedResult {
```
Comment (gloriousCode) [inline currency/pair_test.go:1061]: Use t.Run()+t.Parallel

File: currency/pair_methods.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 // EMPTYFORMAT defines an empty pair format
 var EMPTYFORMAT = PairFormat{}
 
-var errCurrencyNotAssociatedWithPair = errors.New("currency not associated with pair")
+var ErrCurrencyNotAssociatedWithPair = errors.New("currency not associated with pair")
```
Comment (gloriousCode) [inline currency/pair_methods.go:12]: Lint

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Looks good! Loving the amount of lines changed 😄

**Review by gloriousCode — APPROVED** [review]
Happy tACKday Shazbert

**Review by thrasher- — APPROVED** [review]
LGTM

**Pull request merged**